corehq/apps/users/views/web.py

```diff
@@ -161,7 +162,8 @@ def __call__(self, request, uuid, **kwargs):
                     send_hubspot_form(HUBSPOT_NEW_USER_INVITE_FORM, request, user)
                     return HttpResponseRedirect(self.redirect_to_on_success(invitation.email, invitation.domain))
             else:
-                if CouchUser.get_by_username(invitation.email):
+                if (CouchUser.get_by_username(invitation.email)
+                        or User.objects.filter(username__iexact=invitation.email).count() > 0):
```

Comment:
I'd prefer that this implementation detail is contained within the `CouchUser` class, preferably with a note explaining this was an edge case we ran into in the past. Otherwise, I think myself and other devs would find this logic pretty puzzling.